app/validators/routes_and_redirects_validator.rb

```diff
@@ -147,11 +147,37 @@ def validate(location, redirect)
 
   private
 
+    def acceptable_destination?(target)
+      target.starts_with?("/") || is_valid_govuk_campaign_url?(target)
+    end
+
+    def is_valid_govuk_campaign_url?(target)
+      uri = URI.parse(target)
+      host = uri.host
+      if (host =~ /\A.+\.campaign\.service\.gov\.uk\z/i) != nil &&
+          %w(http https).include?(uri.scheme)
+        label = host&.split(".")&.first
+        label.present? && is_subdomain?(label)
+      end
+    rescue
+      false
+    end
+
+    def is_subdomain?(label)
+      return false unless (1..63) === label.length
```

Comment:
I mean can we make it more explicit. Without the context of the PR, I wouldn't know what the magic number 63 means. So you could pull the `(1..63)` into a method that explains that it is the DNS label range, or similar.